crates/rpc/rpc-types/src/eth/transaction/mod.rs

```diff
@@ -152,4 +152,38 @@ impl Transaction {
             transaction_type: Some(U64::from(signed_tx.tx_type() as u8)),
         }
     }
+
+    /// Create a new rpc transaction result for a pending signed transaction, setting block
+    /// environment related fields to `None`.
+    pub fn into_transaction(self) -> TransactionSigned {
+        let to = match self.to {
+            None => TransactionKind::Create,
+            Some(to) => TransactionKind::Call(to),
+        };
+
+        let transaction = match self.transaction_type.map(|x| x.as_u64()) {
+            None | Some(0) => PrimitiveTransaction::Legacy(TxLegacy {
+                chain_id: self.chain_id.map(|x| x.as_u64()),
```

Comment:
This needs to default to 1 otherwise it fails, the geth types do not have chain ID in `txs.json`.